### PR TITLE
Dev v5

### PR DESCRIPTION
**Pull request by pastika**

### CODE REVIEW

**Comment by zhenbinwu** [comment]
Let us use this dev_v5 branch for the v5 ntuple production.

File: python/modules/SoftBDeepAK8SFProducer.py [hunk 1]
```diff
@@ -187,6 +184,43 @@ def GetDeepAK8SF(self, fjets):
...
+            GenPart_statusFlags = np.fromiter(self.TTreeReaderArrayWrapper(event.GenPart_statusFlags), int)
+            GenPart_eta = np.fromiter(self.TTreeReaderArrayWrapper(event.GenPart_eta), float)
+            GenPart_phi = np.fromiter(self.TTreeReaderArrayWrapper(event.GenPart_phi), float)
+
+            fatJet_eta = np.fromiter(self.TTreeReaderArrayWrapper(event.FatJet_eta), float)
+            fatJet_phi = np.fromiter(self.TTreeReaderArrayWrapper(event.FatJet_phi), float)
+
+            # statusFlag 0x2100 corresponds to "isLastCopy and fromHardProcess"
+            # statusFlag 0x2080 corresponds to "IsLastCopy and isHardProcess"
+            genPartsFilt = (((abs(GenPart_pdgId) >= 1) & (abs(GenPart_pdgId) <= 5)) | (abs(GenPart_pdgId) == 21)) & (((GenPart_statusFlags & 0x2100) == 0x2100) | ((GenPart_statusFlags & 0x2080) == 0x2080))
```
Comment (zhenbinwu) [inline python/modules/SoftBDeepAK8SFProducer.py]: Here the nGenPart doesn't need to come from top quark? Can we store this variable per fatjet? Might be useful for other study.

Comment (pastika) [reply python/modules/SoftBDeepAK8SFProducer.py]: the mass doesn't care if the parton came from a top or not, so we make no such requirement here

Comment (zhenbinwu) [reply python/modules/SoftBDeepAK8SFProducer.py]: OK, do we want to store the nGenPart per fatjet?

Comment (pastika) [reply python/modules/SoftBDeepAK8SFProducer.py]: not a bad idea, I will push a change to add this

**Comment by zhenbinwu** [comment]
Ok, this branch become the official dev_v5 branch for production. So I am closing this PR

**Pull request closed without merging**